### DISCUSSION TITLE
how the webpack works

### DISCUSSION THREAD

**Discussion by lizuncong** [webpack/webpack#14933] [General]
Hello, I was looking at the webpack source code recently, and I feel that the webpack source code is very complicated. Is there a related principle introduction article? Thanks!

**Comment by alexander-akait** [comment]
Could you clarify what specific things are difficult and what do you want to know?

**Reply by lizuncong** [reply]
Thank you for your reply. I just finished reading the webpack main process source code, and spent a month to implement it myself.
https://github.com/lizuncong/mini-webpack

**Reply by alexander-akait** [reply]
Good job :star:
